tools/npm/npm1.md

```diff
@@ -0,0 +1,158 @@
+# Definition
+
+### What is npm?
+**npm** is the **node package manager** that makes it easy for JavaScript developers to share the code that they've created to solve particular problems, and for other developers to reuse that code in their own applications.
+
+### What are node package?
+Node package is a package (source code) that implements a functionality and can be imported and used in a project.
+
+### Where can i find the node packages?
+**npm** allows you to install node packages from [Registry](https://www.npmjs.com/). These are open source packages. Since they are open source is best practice to check the **Stats** and **stars** in **GitHub** before using them.
+
+# Commands
+
+### Installing **npm**
+**npm** is installed as part of the **node.js**
+
+Check **node** version 
+```sh
+node -v
+```
+
+Check **npm** version
+```sh
+npm -v
+```
+
+### Initialialize npm
+
+A process will be started that will create **package.json** file in your current folder. A questionary command line is initiated where at least **name** and **version** should be provided.
+```sh
+npm init
+```
+
+A **package.json** file serves as documentation for what packages your project depends on. It also allows you to specify the versions of a package that your project can use using [semantic versioning rules](http://semver.org/).
+
+### Install a package locally
+
+Install the **dependencies** and **devDependencies** dev read from **package.json** in the local **node_modules** folder.
+```sh
+npm install
+```
+
+In Production there is a flag that installs only **dependencies**.
+```sh
+npm install -production
+```
+
+A package can be downloaded with the command
+```sh
+npm install <package_name>
+```
+This will create the **node_modules** directory in your current directory(if one doesn't exist yet), and will download the package to that directory.
+
+##### Example A
+```sh
+npm install react-native-cli -g
+```
+**npm** will install package **react-native-cli** as a computer program. The package is stored in a global path on your system.
+> **-g** flag: global install flag, if used **react-native-cli** package can run from any path
```

Comment:
this markdown is not very well visible in view mode when i read the file as a whole. Maybe try something different?